src/processor/InstanceProcessor.ts

```diff
@@ -24,9 +26,48 @@ export class InstanceProcessor {
     }
   }
 
-  static process(input: any, implementationGuide: any): ExportableInstance {
+  static extractRules(input: any, target: ExportableInstance, fisher: utils.Fishable): void {
+    const newRules: ExportableInstance['rules'] = [];
+    // Clone input so it can be modified
+    const inputJSON = cloneDeep(input);
+    const instanceOfJSON = fisher.fishForFHIR(
+      target.instanceOf,
+      utils.Type.Resource,
+      utils.Type.Profile,
+      utils.Type.Extension,
+      utils.Type.Type
+    );
+
+    if (instanceOfJSON == null) {
+      logger.error(
+        `InstanceOf definition not found for ${input.id} and cannot export any Assignment Rules.`
+      );
+      return;
+    }
+
+    IGNORED_PROPERTIES.forEach(prop => {
+      delete inputJSON[prop];
+    });
+    if (inputJSON.text?.status === 'generated') {
+      delete inputJSON.text;
+    }
+
+    const flatInstance = getPathValuePairs(inputJSON);
+    const flatInstanceOf = getPathValuePairs(instanceOfJSON);
+    Object.keys(flatInstance).forEach(key => {
+      if (flatInstanceOf[key] == null || !isEqual(flatInstance[key], flatInstanceOf[key])) {
```

Comment:
I don't think this check is having its intended effect. `flatInstance` is an Instance, and `flatInstanceOf` is a StructureDefinition, so `flatInstanceOf` wouldn't have a field at `key`. `flatInstanceOf` would have a snapshot which describes the elements in `flatInstance`, but I think this comparison is mixing the worlds of Instances and StructureDefinitions.

I don't think it is necessary to have a check like this at all. If something is set on the instance, and it isn't captured by a keyword, then I think we want to make a rule for setting that value. Certain elements on the StructureDefinition may be `1..1` and have a `fixed` or `pattern` value set, but the rules around which values SUSHI can infer are complicated, and I don't like the idea of trying to recreate those inference rules on GoFSH.

It's possible we could just take the logic directly from SUSHI, but if we do that, we should do it in an optimizer anyway, and I don't think it is within scope of this task.

Reply:
I think you're right. I think I had changed what `flatInstanceOf` was based off of a few times and hadn't reconsidered where it ended up. I took out the check in 76b4903.